lib/ansible/vars/manager.py

```diff
@@ -429,6 +429,10 @@ def plugins_by_groups():
         # if we have a task and we're delegating to another host, figure out the
         # variables for that host now so we don't have to rely on hostvars later
         if task and task.delegate_to is not None and include_delegate_to:
+            # connection variables must not be propagated to the delegated host
+            for connection_var in C.COMMON_CONNECTION_VARS:
```

Comment:
several things wrong with this:
- the correct value might not be available here as per plugin specific vars would override, ansible_aws_user > ansible_user.
- if the delegated host has no corresponding variable defined, this will leave the common vars undefined
- we are already setting the correct values in task_executor for the connection plugin, this could only affect 'display/templating' but there are cases in which you want the information from the original host (inventory_hostname) .. like when you template for this host information on a firewall (delegate_to).